plugins/modules/datahub_definition_info.py

```diff
@@ -0,0 +1,159 @@
+author:
+  - "Webster Mudge (@wmudge)"
+  - "Dan Chaffelson (@chaffelson)"
+requirements:
+  - cdpy
+options:
+  name:
+    description:
+      - If a name is provided, that Definition will be described.
+      - If no name provided, all Definitions will be listed.
+    type: str
+    required: False
+    aliases:
+      - definition
+extends_documentation_fragment:
+  - cloudera.cloud.cdp_sdk_options
+  - cloudera.cloud.cdp_auth_options
+'''
+
+EXAMPLES = r'''
+# Note: These examples do not set authentication details.
+
+# List basic information about all Datahubs
+- cloudera.cloud.datahub_definition_info:
+
+# Gather detailed information about a named Datahub
+- cloudera.cloud.datahub_definition_info:
+    name: example-definition
+'''
+
+RETURN = r'''
+---
+definitions:
+  description: The information about the named Definition or Definitions
+  type: list
+  returned: on success
+  elements: complex
+  contains:
+    clusterDefinitionName:
+      description: The name of the cluster definition.
+      returned: always
+      type: str
+    crn:
+      description:  The CRN of the cluster definition.
+      returned: always
+      type: str
+    description:
+      description: The description of the cluster definition.
+      returned: always
+      type: str
+    productVersion:
+      description: The product version.
+      returned: always
+      type: str
+    instanceGroupCount:
+      description: The instance group count of the cluster.
+      returned: always
+      type: str
+    status:
+      description: The status of the cluster definition.
+      returned: always
+      type: str
+    tags:
+      description: Tags added to the cluster definition
+      type: dict
+      returned: always
+      contains:
+        key:
+          description: The key of the tag.
+          returned: always
+          type: str
+        value:
+          description: The value of the tag.
+          returned: always
+          type: str
+sdk_out:
+  description: Returns the captured CDP SDK log.
+  returned: when supported
+  type: str
+sdk_out_lines:
+  description: Returns a list of each line of the captured CDP SDK log.
+  returned: when supported
+  type: list
+  elements: str
+'''
+
+
+class DatahubDefinitionInfo(CdpModule):
+    def __init__(self, module):
+        super(DatahubDefinitionInfo, self).__init__(module)
+
+        # Set variables
+        self.name = self._get_param('name')
+
+        # Initialize return values
+        self.definitions = []
+
+        # Execute logic process
+        self.process()
+
+    @CdpModule._Decorators.process_debug
+    def process(self):
+        if self.name:  # Note that both None and '' will trigger this
+            definition_single = self.cdpy.datahub.describe_defintion(self.name)
+            if definition_single is not None:
+                self.definitions.append(definition_single)
+        else:
+            self.definitions = self.cdpy.datahub.list_cluster_definitions()
```

Comment:
Return values need to be normalized, so in this case, the "basic" values returned by `list_cluster_definitions` need to be expanded -- sadly, we need to do this via brute force and iterate over the list of returned definitions.

Reply:
I don't know what this means...

Reply:
I copied "datahub_template_info.py" to create this.

Reply:
Oh, look at! Also not good. Thanks for pointing that out!

Reply:
I think by normalising the output of two different commands _when we don't actually need to_ is just introducing additional maintenance overhead in future if the upstream CLI changes. When are you going to need to deal with them having the same return. I can't see a situation where this would be necessary in practical usage. Better to request a change upstream to have the list and describe command return the same fields

Reply:
Are you suggesting that we have two different `_info` modules? The whole point of the `_info` module is to normalize the return values. I agree that we should request an upstream change, esp. since the delta between the two commands is minimal, but I don't understand your comment "we don't need to."